lib/BackgroundTaskManager.js

```diff
@@ -46,39 +45,24 @@ module.exports = kind.singleton(
 	/**
 	* @private
 	*/
+	namedCustomers: {},
+
+	/**
+	* @private
+	*/
+	currentIndex: 0,
+
+	/**
+	* @private
+	*/
 	constructor: kind.inherit( function (sup) {
 		return function () {
-			var c, p, f = 0, d = 1000 / this.fpsThreshold;
-
 			sup.apply(this, arguments);
 
-			this.cb = this.bindSafely(function() {
-				if (this.customers.length) {
-					if (!SystemMonitor.active) {
-						SystemMonitor.trigger();
-					}
-
-					if (!c) {
-						c = utils.perfNow();
-					} else {
-						p = c;
-						c = utils.perfNow();
-						f = ((c - p) < d) ? f + 1 : 0;
-					}
-					if (f == this.frameThreshold && SystemMonitor.idle()) {
-						this.run();
-						c = p = f = 0;
-					} else {
-						// reset fps check if threshold is met but system is not user-idle
-						if (f == this.frameThreshold) {
-							c = p = f = 0;
-						}
-						this.trigger();
-					}
-
-				} else if (SystemMonitor.active) {
-					SystemMonitor.stop();
-				}
+			SystemMonitor.on('idle', this.notifyIdle, this);
+
+			this.cb = this.bindSafely( function () {
+				if (!SystemMonitor.get('active')) SystemMonitor.start();
```

Comment:
Doesn't appear you need to bind the function since you're not using any local context. Assuming that's just cruft from the previous code. Also, since `SystemMonitor.start()` has a check already for `active`, that's probably not necessary here. I think that would make more sense anyway as it's probably not the caller's responsibility to know if SystemMonitor is currently active before calling `start()`.